src/seedu/addressbook/AddressBook.java

```diff
@@ -308,11 +308,20 @@ private static void setupDefaultFileForStorage() {
      * Returns true if the given file is acceptable.
      * The file path is acceptable if it ends in '.txt'
      * TODO: Implement a more rigorous validity checking.
+     * Note: Access modifier was changed from private to 
+     * protected so as to test the method.
      */
-    private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+    protected static boolean isValidFilePath(String filePath) {
+        if (filePath == null) {
+            return false;
+        }
+        File fileFromPath = new File(filePath);     
+        if (fileFromPath.isDirectory()) {
+            return false;
+        }
+        return fileFromPath.exists();
     }
```

Comment:
If this is how the final method is going to be, then it is no longer specific to AddressBook class and can be moved to a utility class.